.github/workflows/changelogs.yml

```diff
@@ -0,0 +1,80 @@
+name: changelogs
+on:
+  push:
+    branches:
+      - '**'
+jobs:
+  update_changelogs:
+    runs-on: ubuntu-20.04
+    outputs:
+      output1: ${{ steps.gitversion.outputs.version }}
+      output2: ${{ steps.date.outputs.date }}
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: setup python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.8'
+    - name: Install package
+      run: pip install python-semantic-release
+
+    - name: Git setup
+      run: |
+        git config --global user.email "actions@github.com"
+        git config --global user.name "github-actions"
+
+    - name: Determine Version
+      id: gitversion
+      run: echo "::set-output name=version::$(semantic-release print-version)"
+
+    - name: Get Latest Tag
+      id: latesttag
+      run: echo "::set-output name=tag::$(git describe --tags --abbrev=0)"
+
+    - name: Update Version
+      if: ${{ !contains(steps.gitversion.outputs.version,steps.latesttag.outputs.tag) }}
```

Comment:
This conditional appears a lot.  What is it checking?

Reply:
It checks whether Gitversion suggests a version upgrade or it points to the already created tag. If a version upgrade is suggested, only then further steps execute.

Reply:
Maybe we can simplify this by using two jobs, and having the second one have one "if" that tests the outputs from the first job?  https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idoutputs